packages/wrangler/src/dev/dev.tsx

```diff
@@ -458,24 +553,30 @@ function DevSession(props: DevSessionProps) {
 		onBundleStart();
 	}, [esbuildStartTimeoutRef, onBundleStart]);
 
-	useCustomBuild(props.entry, props.build, onBundleStart, onCustomBuildEnd);
-
 	const directory = useTmpDir(props.projectRoot);
 
-	const workerDefinitions = useDevRegistry(
-		props.name,
-		props.bindings.services,
-		props.bindings.durable_objects,
-		props.local ? "local" : "remote"
-	);
 	useEffect(() => {
 		// temp: fake these events by calling the handler directly
 		devEnv.proxy.onConfigUpdate({
 			type: "configUpdate",
 			config: startDevWorkerOptions,
 		});
-	}, [devEnv, startDevWorkerOptions]);
+		if (props.experimentalDevEnv) {
+			void devEnv.bundler.onConfigUpdate({
+				type: "configUpdate",
+				config: startDevWorkerOptions,
+			});
+		}
+	}, [devEnv, startDevWorkerOptions, props.experimentalDevEnv]);
 
+	// This value is static, so doesn't cause the below hooks to be called conditionally
+	if (props.experimentalDevEnv) {
+		return null;
+	}
```

Comment:
In the interest of derisking, we may want to be more granular with the if-block. See my comments below

Reply:
Given the long term goal is to move away from React, is this definitely a direction we want to go in? It feels like starting to move away from React with things like rendering the account choices list is the right direction, but I'm happy to defer that to a later stage if you think that's too risky